packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -69,129 +74,191 @@ describe("R11s Socket Tests", () => {
-			mockSocket(socket as unknown as Socket, async () =>
-				documentService.connectToDeltaStream(client),
-			),
-			{
-				errorType: RouterliciousErrorTypes.clusterDrainingError,
-				scenarioName: "connect_document_error",
+	// Socket error tests
+	const socketErrorTests = [
+		{
+			description: "Socket error with Token Revoked error",
+			errorToThrow: {
+				code: 403,
+				message: "TokenRevokedError",
+				retryAfterMs: 10,
+				internalErrorCode: "TokenRevoked",
+				errorType: DriverErrorTypes.authorizationError,
+				canRetry: false,
+			},
+			expectedErrorType: DriverErrorTypes.authorizationError,
+			expectedInternalErrorCode: "TokenRevoked",
+		},
+		{
+			description: "Socket error with Cluster Draining error",
+			errorToThrow: {
+				code: 503,
+				message: "ClusterDrainingError",
+				retryAfterMs: 1000,
 				internalErrorCode: R11sServiceClusterDrainingErrorCode,
+				errorType: RouterliciousErrorTypes.clusterDrainingError,
+				canRetry: true,
 			},
-			"Error should have occurred",
-		);
-	});
-
-	it("Socket error with Cluster Draining error", async () => {
-		const errorToThrow = {
-			code: 503,
-			message: "ClusterDrainingError",
-			retryAfterMs: 1000,
-			internalErrorCode: R11sServiceClusterDrainingErrorCode,
-			errorType: RouterliciousErrorTypes.clusterDrainingError,
-			canRetry: true,
-		};
-		const errorEventName = "connect_document_success";
-		socket = new ClientSocketMock({
-			connect_document: { eventToEmit: errorEventName },
-		});
+			expectedErrorType: RouterliciousErrorTypes.clusterDrainingError,
+			expectedInternalErrorCode: R11sServiceClusterDrainingErrorCode,
+		},
+	];
 
-		const connection = await mockSocket(socket as unknown as Socket, async () =>
-			documentService.connectToDeltaStream(client),
-		);
-		let error: IAnyDriverError | undefined;
-		connection.on("disconnect", (reason: IAnyDriverError) => {
-			error = reason;
-		});
+	socketErrorTests.forEach((test) =>
+		runSocketErrorTest(
+			test.description,
+			test.errorToThrow,
+			test.expectedErrorType,
+			test.expectedInternalErrorCode,
+		),
+	);
 
-		// Send Token Revoked error
-		socket.sendErrorEvent(errorToThrow);
+	// Client error tests
+	runClientErrorTest(
+		"Client Data Corruption error",
+		new DataCorruptionError("Data corruption error", { driverVersion: "1.0" }),
+		FluidErrorTypes.dataCorruptionError,
+	);
 
-		assert(
-			error?.errorType === RouterliciousErrorTypes.clusterDrainingError,
-			"Error type should be clusterDrainingError",
-		);
-		assert(error.scenarioName === "error", "Error scenario name should be error");
-		assert(
-			(error as any).internalErrorCode === R11sServiceClusterDrainingErrorCode,
-			"Error internal code should be R11sServiceClusterDrainingErrorCode",
-		);
-	});
+	runClientErrorTest(
+		"Client Data Processing error",
+		DataProcessingError.create("Data processing error", "test", undefined, {
+			driverVersion: "1.0",
+		}),
+		FluidErrorTypes.dataProcessingError,
+	);
```

Comment:
Let's add a test that checks that no event is fired if there's no error that caused teh disconnect.